pkg/controller/operators/catalog/operator.go

```diff
@@ -1331,10 +1331,44 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 		return
 	}
 
+	// Complete and Failed are terminal phases
+	if plan.Status.Phase == v1alpha1.InstallPlanPhaseFailed || plan.Status.Phase == v1alpha1.InstallPlanPhaseComplete {
```

Comment:
This makes sense logically but looking at https://issues.redhat.com/browse/OLM-2114 coming in as well -- if there is a transient error in `ExecutePlan()` we will not mark the installplan as `Failed` but keep it in the `Installing` phase for a certain period of time right?

The new behavior would be that in the `ExecutePlan()` function that runs the installplan steps any error returned from that function effectively causes installplan reconciliation to stop. We do not have that behavior today (the installplan stays in the `Failed` state but continues to be reconciled). 

The reason I bring this up is because of https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/scoped-operator-install.md#when-operator-install-fails
I believe we have a feature where if the ServiceAccount used to install the operator lacks certain permissions, we encourage the user to modify the SA to get to the permission level they need and OLM will automatically requeue up the installplan. This changes that behavior. cc @ecordell if I'm mistaken.

Reply:
> the installplan stays in the Failed state but continues to be reconciled

If it's in Failed, then when it reconciles again it hits the default case in `transitionInstallPlanState` and remains Failed. The in-progress retry feature is going to prevent the transition from Installing -> Failed and requeue as long as a timeout hasn't been reached.

Reply:
Ok. I seem to recall the scenario I described above as desirable (if the SA isn't quite right it can be modified and the InstallPlan can eventually succeed from a conversation with Evan). But it does seem like an anti-pattern in general so I am ok moving ahead as-is.